thunx-autoconfigure/src/main/java/com/contentgrid/thunx/gateway/autoconfigure/GatewayAutoConfiguration.java

```diff
@@ -49,19 +55,22 @@ public OpaInputProvider<Authentication, ServerWebExchange> defaultOpaInputProvid
 
     @Bean
     @ConditionalOnMissingBean
+    @ConditionalOnBean({OpaClient.class})
     public PolicyDecisionPointClient<Authentication, ServerWebExchange> pdpClient(OpaClient opaClient, OpaQueryProvider<ServerWebExchange> queryProvider, OpaInputProvider<Authentication, ServerWebExchange> inputProvider) {
         return new OpenPolicyAgentPDPClient<>(opaClient, queryProvider, inputProvider);
     }
 
     @Bean
     @ConditionalOnMissingBean
+    @ConditionalOnBean(PolicyDecisionPointClient.class)
     public ReactiveAuthorizationManager<AuthorizationContext> reactiveAuthenticationManager(
             PolicyDecisionPointClient<Authentication, ServerWebExchange> pdpClient) {
         return new ReactivePolicyAuthorizationManager(new PolicyDecisionComponentImpl<>(pdpClient));
     }
 
     @Bean
     @ConditionalOnMissingBean
+    @ConditionalOnBean(PolicyDecisionPointClient.class)
```

Comment:
This isn't really dependent on having a PDP Client?

However, when there isn't a PDP client, it doesn't send an abac context header, which might result in having too broad access.

When disabling this bean, I think it will cause an error during configuration IF an abac filter is present on a route?

Reply:
This is dependent on _something_ putting a `ABAC_POLICY_PREDICATE` in the exchange attributes. This _could_ be another kind of authorization-bean.

If it is not set, it's a no-op, so there is no real impact.

Conclusion: I'm going to remove the `@ConditionalOnBean`